eZ/Publish/API/Repository/Tests/SearchServiceFulltextTest.php

```diff
@@ -100,89 +100,164 @@ public function testPrepareContent()
      */
     public function providerForTestFulltextSearch()
     {
-        return [
-            [
-                'fox',
-                [3, [6, 8, 10], [11, 13, 14, 15]],
-            ],
-            [
-                'quick fox',
-                $quickOrFox = [6, [11, 13, 15], [1, 3], [5, 7, 8, 10], [12, 14]],
-            ],
-            [
-                'quick OR fox',
-                $quickOrFox,
-            ],
-            [
-                'quick AND () OR AND fox',
-                $quickOrFox,
-            ],
-            [
-                '+quick +fox',
-                $quickAndFox = [6, [11, 13, 15]],
-            ],
-            [
-                'quick AND fox',
-                $quickAndFox,
-            ],
-            [
-                'brown +fox -news',
-                [8, 11, 3, 6],
-            ],
-            [
-                'quick +fox -news',
-                [6, 11, 3, 8],
-            ],
-            [
-                'quick brown +fox -news',
-                $notNewsFox = [11, [6, 8], 3],
-            ],
-            [
-                '((quick AND fox) OR (brown AND fox) OR fox) AND NOT news',
-                $notNewsFox,
-            ],
-            [
-                '"quick brown"',
-                [5, [11, 12, 15]],
-            ],
-            [
-                '"quick brown" AND fox',
-                [[11, 15]],
-            ],
-            [
-                'quick OR brown AND fox AND NOT news',
-                [11, 8],
-            ],
-            [
-                '(quick OR brown) AND fox AND NOT news',
-                [11, [6, 8]],
-            ],
-            [
-                '"fox brown"',
-                [],
-            ],
-            [
-                'qui*',
-                [[1, 5, 6, 7, 11, 12, 13, 15]],
-            ],
-            [
-                '+qui* +fox',
-                [6, [11, 13, 15]],
-            ],
-        ];
+        $solrVersion = getenv('SOLR_VERSION');
+
+        if ($solrVersion >= 7) {
```

Comment:
Nooo.... this will be a nightmare to browse through and maintain. What are the differences anyway?

Reply:
> Nooo.... this will be a nightmare to browse through and maintain.

The problem could be solved by moving Solr related tests to `ezplatform-solr-bundle` (as follow up) or we could have separate data providers for Solr 6 and 7 if this will improve readability in your opinion. 

> What are the differences anyway?

Scoring has been slightly changed between 6.X and 7.X so the results grouping differs.

Reply:
> > Nooo.... this will be a nightmare to browse through and maintain.
> 
> The problem could be solved by moving Solr related tests to `ezplatform-solr-bundle` (as follow up) or we could have separate data providers for Solr 6 and 7 if this will improve readability in your opinion.

The former one is probably better, but I can live with the latter one :+1: